metapool/tests/test_rescale_counts.py

```diff
@@ -0,0 +1,136 @@
+                            start_ratio * len2/len1)
+
+    def test_abundance_read_count(self):
+        # Load required data
+        table_community = pd.read_csv(
+            _folder + "table_community_hits.txt",
+            sep="\t", index_col="OTUID")
+        linear_models = pd.read_csv(
+            _folder + "results_synDNA_linear_model_variables.txt",
+            sep="\t")
+        metadata_features = pd.read_csv(
+            _folder + "metadata_features.tsv",
+            sep="\t", index_col="OTUID")
+
+        # Clean sample names (Trims everything after last underscore)
+        linear_models["sample_name"] = \
+            linear_models["sample_name_pool"].replace(
+                "_[^_]+$", "", regex=True)
+
+        df1, failed_cols = to_absolute_abundance_read_count(
+            table_community, linear_models)
+        df1_cell_count = convert_read_count_to_cell_count(
+            df1, metadata_features)
+        df2, failed_cols = to_absolute_abundance_cell_count(
+            table_community, linear_models, metadata_features)
+
+        self.assertTrue(dataframes_are_equal(
+            df1_cell_count, df2, verbose=False))
+
+    def test_normalizations(self):
+        df = np.random.rand(10, 10)
+        df = pd.DataFrame(df)
+        df_col_pct = _to_column_percentage(df)
+        self.assertTrue(
+            ((df_col_pct.sum() > 99.99) &
+             (df_col_pct.sum() < 100.01)).all())
+
+        df_row_pct = _to_row_percentage(df)
+        self.assertTrue(
+            ((df_row_pct.sum(axis=1) > 99.99) &
+             (df_row_pct.sum(axis=1) < 100.01)).all())
+
+    def test_same_as_R(self):
+        # Load required data
+        table_community = pd.read_csv(
+            _folder + "table_community_hits.txt",
+            sep="\t", index_col="OTUID")
+        linear_models = pd.read_csv(
+            _folder + "results_synDNA_linear_model_variables.txt",
+            sep="\t")
+        metadata_features = pd.read_csv(
+            _folder + "metadata_features.tsv",
+            sep="\t", index_col="OTUID")
+
+        # Clean sample names (Trims everything after last underscore)
+        linear_models["sample_name"] = \
+            linear_models["sample_name_pool"].replace(
+                "_[^_]+$",
+                "",
+                regex=True)
+
+        df, failed_samples = to_absolute_abundance_cell_count(
+            table_community, linear_models, metadata_features)
+        # R version applies absolute abundance transform,
+        # cell count transform, and a transpose,
+        # Python version you must externally transpose.
+        df = df.T
+
+        print("WARNING: The following samples don't have "
+              "applicable linear models:", failed_samples)
+        # If you want to check with external tools:
+        # df.to_csv("feature_table_cell_counts.txt", sep="\t")
+        df_r = pd.read_csv(
+            _folder + "feature_table_cell_counts_R.txt",
```

Comment:
Should this be read as a biom vs. a txt? In other words, should the output be a biom table vs. a txt?

Reply:
I do not know if any of the necessary input files exist or in what formats.  I expect it will be necessary to adapt these file types, and possibly remove the dilution file if it is determined that the dilutions are hard coded.

Reply:
All file I/O is done in the test functions, so it is completely fine to change how dataframes are stored in the actual implementation that connects to real data.